src/wallet/wallet.cpp

```diff
@@ -5371,9 +5380,8 @@ bool CWallet::SelectCoinsMinConf(const CAmount& nTargetValue, int nConfMine, int
 bool CWallet::SelectCoins(const CAmount& nTargetValue, set<pair<const CWalletTx*,unsigned int> >& setCoinsRet, CAmount& nValueRet,  bool& fOnlyCoinbaseCoinsRet, bool& fNeedCoinbaseCoinsRet, const CCoinControl* coinControl) const
 {
     // Output parameter fOnlyCoinbaseCoinsRet is set to true when the only available coins are coinbase utxos.
-    vector<COutput> vCoinsNoCoinbase, vCoinsWithCoinbase;
-    AvailableCoins(vCoinsNoCoinbase, std::nullopt, true, coinControl, false, false);
-    AvailableCoins(vCoinsWithCoinbase, std::nullopt, true, coinControl, false, true);
+    auto vCoinsNoCoinbase = AvailableCoins(std::nullopt, true, coinControl, false, false, ISMINE_LEGACY_SPENDABLE);
+    auto vCoinsWithCoinbase = AvailableCoins(std::nullopt, true, coinControl, false, true, ISMINE_LEGACY_SPENDABLE);
```

Comment:
The old code defaulted to `fOnlySpendable = false` here, and now it filters on `ISMINE_LEGACY_SPENDABLE`, so `vCoins*` now won't contain any of the outputs where previously `out.fSpendable = false`. Inside this function all iteration over `vCoins*` filters on `out.fSpendable` so they aren't affected, but `vCoins` below is passed into `SelectCoinsMinConf`, and I'm not sure whether the behaviour of that function requires `fSpendable` to be accurate.